assets/js/search-page.js

```diff
@@ -27,30 +28,85 @@ function initialize () {
   const pathname = window.location.pathname
   if (pathname.endsWith('/search.html') || pathname.endsWith('/search')) {
     const query = getQueryParamByName('q')
-    search(query)
+    const queryType = getQueryParamByName('type')
+    search(query, queryType)
   }
 }
 
-async function search (value) {
+async function search (value, queryType) {
   if (isBlank(value)) {
     renderResults({ value })
   } else {
     setSearchInputValue(value)
 
-    const index = await getIndex()
-
     try {
-      // We cannot match on atoms :foo because that would be considered
-      // a filter. So we escape all colons not preceded by a word.
-      const fixedValue = value.replaceAll(/(\B|\\):/g, '\\:')
-      const results = searchResultsToDecoratedSearchItems(index.search(fixedValue))
+      let results = []
+      const searchNodes = getSearchNodes()
+
+      if (['related', 'latest'].includes(queryType) && searchNodes.length > 0) {
+        results = await remoteSearch(value, queryType, searchNodes)
+      } else {
+        results = await localSearch(value)
+      }
+
       renderResults({ value, results })
     } catch (error) {
       renderResults({ value, errorMessage: error.message })
     }
   }
 }
 
+async function localSearch (value) {
+  const index = await getIndex()
+
+  // We cannot match on atoms :foo because that would be considered
+  // a filter. So we escape all colons not preceded by a word.
+  const fixedValue = value.replaceAll(/(\B|\\):/g, '\\:')
+  return searchResultsToDecoratedSearchItems(index.search(fixedValue))
+}
+
+async function remoteSearch (value, queryType, searchNodes) {
+  let filterNodes = searchNodes
+
+  if (queryType === 'latest') {
+    filterNodes = searchNodes.slice(0, 1)
+  }
+
+  const filters = filterNodes.map(node => `package:=${node.name}-${node.version}`).join(' || ')
```

Comment:
Unless we start using custom URLs like the one mentioned in https://github.com/hexpm/hexdocs/issues/49#issuecomment-2583622344, an array filter like this might help keep the URL shorter (no `package:=` repetition):

```js
if (nodes && nodes.length > 0) {
  const packages = nodes.map((node) => `${node.name}-${node.version}`);
  params.filter_by = `package:=[${packages.join(",")}]`;
}
```

Adapted from Typesense POC: https://gist.github.com/ruslandoga/d544addc4a17e1f4853d7e9ae97818a4